File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        maxP = max(piles)
        res = maxP
        l, r = 1, res
        while l <= r:
            k = l + (r-l) // 2
            hours = 0
            for p in piles:
                hours += math.ceil(p / k)
            if hours <= h:
                res = min(res, k)
                r = k-1
            else:
                l = k+1
        return res
        

File: Data Structures & Algorithms/pow-x-n/submission-2.cpp
class Solution {
public:
    double myPow(double x, int n) {
    if (x == 0)
      return 0;
    if (n == 0)
      return 1;
    if (n == 1) {
      return x;
    }
    double res = myPow(x, abs(n / 2));
    if (n % 2 == 0) {
      res = res * res;
    } else {
      res = res * res * x;
    }
    return (n >= 0) ? res : 1 / res;
  }


};
